[docs] Add a page on triggering EAS builds on the CI (#10895)
expo/expo: docs/pages/build/building-on-ci.md

---
title: EAS builds with CI
---

This document outlines how to trigger builds on EAS for your app from a CI environment.

## Configuring the app

To trigger EAS builds from a CI environment, we first need to configure our app with EAS Build and make sure we can trigger builds from our local machine.
To automatically configure your native project for building with EAS Build on Android and iOS, you will need to run the following command in the root of your project:

```sh
expo eas:build:init
```

See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for a detailed explanation and examples of what this does.

## Triggering builds

Before building with EAS on CI, we need to install and configure `expo-cli`. Then, we can trigger new builds with the `eas:build` command.

### Prepare Expo CLI

To interact with the Expo API, we need to install Expo CLI. You can use an environment with this library preinstalled, or you can add it to the project as a development dependency.

> The later is the easiest way but might increase the installation time.

Reviewer: ```suggestion
The latter is the easiest way, but may increase the installation time.
```

---

> title: EAS builds on the CI

Reviewer: ```suggestion
title: EAS builds with CI
```

---

This document outlines how to trigger builds on EAS for your app from a CI environment.

## Configuring the app

> To be able to trigger builds on EAS from the CI environment, first, we need to ensure that our app is configured to build on EAS and that we can trigger builds from our local machine.

Reviewer: ```suggestion
To trigger EAS builds from a CI environment, we first need to configure our app with EAS Build and make sure we can trigger builds from our local machine.
```

To automatically configure your native project for building with EAS Build on Android and iOS, you will need to run the following command in the root of your project:

```sh
expo eas:build:init
```

> See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for detailed explanation and examples of what this does.

Reviewer: ```suggestion
See [EAS Build from scratch in 5 minutes](../eas-build-in-5-minutes/) for a detailed explanation and examples of what this does.
```

## Triggering builds

> To start triggering EAS builds on the CI, we need to install and configure `expo-cli` and then trigger new builds with the `eas:build` command.

Reviewer: ```suggestion
Before building with EAS on CI, we need to install and configure `expo-cli`. Then, we can trigger new builds with the `eas:build` command.
```

### Prepare Expo CLI

> To interact with the Expo API, we need to install the Expo CLI.

Reviewer: ```suggestion
To interact with the Expo API, we need to install Expo CLI.
```

You can use an environment with this library preinstalled, or you can add it to the project as a development dependency.
The latter is the easiest way but might increase the installation time.
For vendors that charge you per minute, it might we worth creating a prebuilt environment.

> To install the Expo CLI into your project, you can execute this script.

Reviewer: ```suggestion
To install Expo CLI in your project, run: 
```

```sh
npm install --save-dev expo-cli
```

### Prepare authentication

> Next, we will configure the publishing step of your application to Expo.

Reviewer: I'm a little confused by this line- running `expo publish` isn't included in this guide?

Before we can do this, we need to authenticate as the owner of the app.
This is possible by storing the username or email with the password in the environment.
Every vendor has implemented their storage mechanism for sensitive data like passwords, although they are very similar.
Most vendors make use of environment variables which are "injected" into the environment and scrubbed from the logs to keep it safe.

To perform the authentication, we will add this script to our configuration:

```sh

> npx expo login -u <EXPO USERNAME> -p <EXPO PASSWORD>

Reviewer 2: We shouldn't recommend login/password auth on CI, `EXPO_TOKEN` is currently the recommended way

Reviewer: we can link them here for that- https://docs.expo.io/accounts/personal/#programmatic-access

```

If you don't want to expose the password in the login script, set the `EXPO_CLI_PASSWORD` environment variable to the password and run the following script instead:

```sh
npx expo login --non-interactive -u <EXPO USERNAME>
```

### Trigger new builds

> After having the CLI library and authentication in place, we can finally create the build step.

Reviewer: ```suggestion
Now that we're authenticated with Expo CLI, we can create the build step.
```
